instrumentation/executors/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/executors/ExecutorAdviceHelper.java

```diff
@@ -106,6 +109,7 @@ public static void cleanUpAfterSubmit(
       exceptions.
        */
       propagatedContext.clear();
+      virtualField.set(task, null);
```

Comment:
maybe a comment here that this is only needed for cleaning up the fallback map?

or alternatively, do we want to just think of this as a "more correct" behavior?

Reply:
I added a comment. I think that with fallback map we need to do this, though ideally the fallback shouldn't be used at all which is what we really need to fix.